airflow/models/xcom_arg.py

```diff
@@ -0,0 +1,155 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Any, Dict, List, Optional, Union
+
+from airflow.exceptions import AirflowException
+from airflow.models.baseoperator import BaseOperator
+from airflow.models.xcom import XCOM_RETURN_KEY
+
+
+class XComArg:
+    """
+    Class that represents a XCom push from a previous operator.
+    Defaults to "return_value" as only key.
+
+    Current implementations supports
+      xcomarg >> op
+      xcomarg << op
+      op >> xcomarg   (by BaseOperator code)
+      op << xcomarg   (by BaseOperator code)
+
+    **Example**:
+        The moment you got result from any op (functional or regular one) you can ::
+            xcomarg = ...
+            my_op = MyOperator()
+            my_op >> xcomarg
+
+    This object can be used in legacy Operators via Jinja:
+    **Example**:
+        You can make this result to be part of any generated string ::
+            xcomarg = ...
+            op1 = MyOperator(my_text_message=f"{xcomarg}")
+            op2 = MyOperator(my_text_message=f"{xcomarg['topic']}")
+
+
+    """
+
+    def __init__(self, operator: BaseOperator, key: Optional[str] = XCOM_RETURN_KEY,
+                 runtime_sub_keys: Optional[List[str]] = None):
```

Comment:
Should we add description of those params to help everyone understand what's going on?